### DISCUSSION TITLE
Budibase Create User API - No User Info Found

### DISCUSSION THREAD

**Discussion by tailorck** [Budibase/budibase#5759] [Help]
I tried using the budibase api to create a user. I need to do this because I realize that if I create users through the app, I have to wait for the user to create their account before I can assign them the roles on the appropriate apps. While using the API I got an error with `No user info found`. I purposely redacted by API Key in the screenshot below, but do you guys see any other errors?

<img width="460" alt="image" src="https://user-images.githubusercontent.com/7613002/167235646-8060f50c-d204-45a0-8ecc-0fd72151299e.png">

**Comment by mjashanks** [comment]
Hey @tailorck - one issue i can see is that your appId is incorrect. It should look like `app_<guid>`

**Reply by tailorck** [reply]
Where do I get this app id?

**Reply by shogunpurple** [reply]
Hi @tailorck - you can find your app ID by opening up the app in your builder and looking at the URL. Your URL will look something like this:
`https://bb.budibase.app/builder/app/app_dev_bb_dba978ad3a8d443397e11d6c7ca6f4e0`

This is the dev URL - to get the appID of your published app, you take away  the `_dev_` from the appID. So in this case it is:

`app_bb_dba978ad3a8d443397e11d6c7ca6f4e0`

I've created a PR to add this option to the `Edit` menu in the app list, which will allow you to just select an option to copy your appID directly for use with the budibase API and make this process easier. 

https://github.com/Budibase/budibase/pull/5771

Thanks!

**Reply by tailorck** [reply]
This is awesome! Thank you, Shogun!

**Reply by tailorck** [reply]
My app id looks like this though `app_dev_b324b516c4764627a3e418c31f4af92c`. Should it be `app_bb_dev_b324b516c4764627a3e418c31f4af92c`?

**Reply by shogunpurple** [reply]
@tailorck You need to publish your app first.

Your app id would then be `app_b324b516c4764627a3e418c31f4af92c`

**Reply by tailorck** [reply]
<img width="462" alt="image" src="https://user-images.githubusercontent.com/7613002/167435497-03e47836-efb1-4d4b-a570-02fcaf00894b.png">

**Reply by tailorck** [reply]
I have published and can open my published app, but this is what my `app_id` shows as when I am in the design mode.

**Reply by tailorck** [reply]
Took the `bb` out in this example and I even got my API key in there: 
<img width="455" alt="image" src="https://user-images.githubusercontent.com/7613002/167437009-f4d286a9-f9ea-4ef0-b553-7a4265ea766b.png">

**Reply by tailorck** [reply]
While I have your attention, can I get some love on this discussion as well: https://github.com/Budibase/budibase/discussions/5739?

**Reply by shogunpurple** [reply]
I was just able to successfully do this in a cloud tenant - I can't see the rest of your configuration, but this worked for me - are you missing something that I might have in my configuration?

![Frame 1(2)](https://user-images.githubusercontent.com/11256663/167443834-fa0a8bfd-01f0-48c3-94ab-c89ec0d31460.png)



Here's my roles too:
<img width="622" alt="Screenshot 2022-05-09 at 16 24 57" src="https://user-images.githubusercontent.com/11256663/167443363-28d5c9ec-c44f-437d-8b82-710a6db5e77c.png">

`No user info` found would indicate that it's trying to load an existing user. I can't seem to recreate this with a bunch of simple user creation configurations
